lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -891,13 +894,51 @@ class ExtraLinesData {
   }
 }
 
-/// Holds data to handle touch events, and touch responses in the [LineChart].
+/// Used for showing touch indicators (a thicker line and larger dot on the targeted spot).
 ///
-/// There is a touch flow, explained [here](https://github.com/imaNNeoFighT/fl_chart/blob/master/repo_files/documentations/handle_touches.md)
-/// in a simple way, each chart captures the touch events, and passes a concrete
-/// instance of [FlTouchInput] to the painter, and gets a generated [LineTouchResponse].
-class LineTouchData extends FlTouchData {
+/// It gives you the [spotIndexes] that touch happened, or manually targeted,
+/// in the given [barData], you should return a list of [TouchedSpotIndicatorData],
+/// length of this list should be equal to the [spotIndexes.length],
+/// each [TouchedSpotIndicatorData] determines the look of showing indicator.
+typedef GetTouchedSpotIndicator = List<TouchedSpotIndicatorData> Function(
+    LineChartBarData barData, List<int> spotIndexes);
 
+List<TouchedSpotIndicatorData> defaultTouchedIndicators(
```

Comment:
I think these changes are unuseful.
It seems you have copied them and paste them in the top.